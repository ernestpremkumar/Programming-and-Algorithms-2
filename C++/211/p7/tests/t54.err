insert <7> failed. String already in tree.
insert <7> failed. String already in tree.
insert <7> failed. String already in tree.
insert <5> failed. String already in tree.
insert <7> failed. String already in tree.
insert <3> failed. String already in tree.
insert <3> failed. String already in tree.
insert <7> failed. String already in tree.
insert <3> failed. String already in tree.
insert <3> failed. String already in tree.
insert <6> failed. String already in tree.
insert <0> failed. String already in tree.
insert <3> failed. String already in tree.
insert <6> failed. String already in tree.
insert <7> failed. String already in tree.
insert <2> failed. String already in tree.
insert <5> failed. String already in tree.
insert <6> failed. String already in tree.
insert <0> failed. String already in tree.
insert <3> failed. String already in tree.
insert <2> failed. String already in tree.
insert <5> failed. String already in tree.
insert <0> failed. String already in tree.
insert <3> failed. String already in tree.
insert <5> failed. String already in tree.
insert <3> failed. String already in tree.
insert <7> failed. String already in tree.
insert <2> failed. String already in tree.
insert <2> failed. String already in tree.
insert <3> failed. String already in tree.
insert <1> failed. String already in tree.
insert <4> failed. String already in tree.
insert <5> failed. String already in tree.
insert <4> failed. String already in tree.
insert <0> failed. String already in tree.
insert <0> failed. String already in tree.
insert <7> failed. String already in tree.
insert <4> failed. String already in tree.
insert <0> failed. String already in tree.
insert <6> failed. String already in tree.
insert <2> failed. String already in tree.
insert <5> failed. String already in tree.
insert <7> failed. String already in tree.
insert <1> failed. String already in tree.
insert <3> failed. String already in tree.
insert <5> failed. String already in tree.
insert <3> failed. String already in tree.
insert <0> failed. String already in tree.
insert <3> failed. String already in tree.
insert <6> failed. String already in tree.
insert <2> failed. String already in tree.
insert <1> failed. String already in tree.
insert <7> failed. String already in tree.
insert <1> failed. String already in tree.
insert <3> failed. String already in tree.
insert <3> failed. String already in tree.
insert <7> failed. String already in tree.
insert <0> failed. String already in tree.
insert <4> failed. String already in tree.
insert <1> failed. String already in tree.
insert <6> failed. String already in tree.
insert <7> failed. String already in tree.
insert <7> failed. String already in tree.
insert <4> failed. String already in tree.
insert <2> failed. String already in tree.
insert <0> failed. String already in tree.
insert <6> failed. String already in tree.
insert <4> failed. String already in tree.
insert <5> failed. String already in tree.
insert <7> failed. String already in tree.
insert <4> failed. String already in tree.
insert <3> failed. String already in tree.
insert <6> failed. String already in tree.
insert <4> failed. String already in tree.
insert <4> failed. String already in tree.
insert <4> failed. String already in tree.
insert <4> failed. String already in tree.
insert <1> failed. String already in tree.
insert <1> failed. String already in tree.
insert <5> failed. String already in tree.
insert <1> failed. String already in tree.
insert <6> failed. String already in tree.
insert <1> failed. String already in tree.
insert <0> failed. String already in tree.
insert <4> failed. String already in tree.
insert <7> failed. String already in tree.
insert <4> failed. String already in tree.
insert <1> failed. String already in tree.
insert <6> failed. String already in tree.
insert <1> failed. String already in tree.
insert <0> failed. String already in tree.
insert <4> failed. String already in tree.
insert <0> failed. String already in tree.
insert <0> failed. String already in tree.
insert <6> failed. String already in tree.
insert <7> failed. String already in tree.
insert <1> failed. String already in tree.
insert <4> failed. String already in tree.
insert <6> failed. String already in tree.
insert <4> failed. String already in tree.
insert <5> failed. String already in tree.
insert <6> failed. String already in tree.
insert <6> failed. String already in tree.
insert <5> failed. String already in tree.
insert <4> failed. String already in tree.
insert <6> failed. String already in tree.
insert <7> failed. String already in tree.
insert <7> failed. String already in tree.
insert <5> failed. String already in tree.
insert <7> failed. String already in tree.
