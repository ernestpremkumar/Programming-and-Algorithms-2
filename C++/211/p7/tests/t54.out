{}
{}
{}
<6> is not in tree.
0
2
{}
2
{}
{}
1
{}
<6> is not in tree.
{}
0
{}
Tree is balanced.
Tree is balanced.
{2}
Average distance of nodes to root = 0
{2}
<6> is not in tree.
{2}
5
<7> is in tree.
Average distance of nodes to root = 0.5
Tree is balanced.
2
Tree is balanced.
6
0
2
2
Average distance of nodes to root = 0.5
Average distance of nodes to root = 0.5
Tree is balanced.
3
0
Tree is balanced.
{2, 7}
<5> is not in tree.
{2, 3, 7}
{2, 3, 7}
Average distance of nodes to root = 1
Average distance of nodes to root = 1
{2, 7, 3}
{2, 3, 7}
0
Average distance of nodes to root = 1
{2, 3, 7}
Average distance of nodes to root = 1.5
{2, 7, 3, 5}
{2, 3, 5, 7}
<2> is in tree.
{2, 7, 3, 5}
{2, 7, 3, 5}
{2, 7, 3, 5}
2
Average distance of nodes to root = 1.5
4
Tree is not balanced.
{2, 3, 5, 7}
Average distance of nodes to root = 1.5
5
<7> is in tree.
{2, 7, 3, 5, 6}
<6> is in tree.
3
{2, 7, 3, 5, 6}
<0> is not in tree.
Tree is not balanced.
Tree is not balanced.
{2, 3, 5, 6, 7}
{2, 7, 3, 5, 6}
Average distance of nodes to root = 2
5
<6> is in tree.
{2, 3, 5, 6, 7}
<4> is not in tree.
Average distance of nodes to root = 2
5
5
<7> is in tree.
5
5
2
{2, 7, 3, 5, 6}
{2, 3, 5, 6, 7}
<4> is not in tree.
5
{2, 7, 3, 5, 6}
5
{2, 3, 5, 6, 7}
Tree is not balanced.
5
{2, 7, 3, 5, 6}
Average distance of nodes to root = 2
<6> is in tree.
{2, 3, 5, 6, 7}
Tree is not balanced.
Average distance of nodes to root = 2
5
Average distance of nodes to root = 2
Tree is not balanced.
{2, 3, 5, 6, 7}
Average distance of nodes to root = 2
Average distance of nodes to root = 2
2
Tree is not balanced.
2
Tree is not balanced.
{2, 7, 3, 5, 6}
{2, 3, 5, 6, 7}
Tree is not balanced.
{2, 0, 7, 3, 5, 6}
6
Tree is not balanced.
6
{0, 2, 3, 5, 6, 7}
6
{0, 2, 3, 5, 6, 7}
{2, 0, 7, 3, 5, 6}
{0, 2, 3, 5, 6, 7}
Tree is not balanced.
{2, 0, 7, 3, 5, 6}
6
<2> is in tree.
6
Average distance of nodes to root = 1.83333
Average distance of nodes to root = 1.83333
4
<0> is in tree.
7
Average distance of nodes to root = 1.83333
{0, 2, 3, 5, 6, 7}
Tree is not balanced.
{2, 0, 7, 3, 5, 6}
{0, 2, 3, 5, 6, 7}
1
Average distance of nodes to root = 1.83333
0
Tree is not balanced.
6
{2, 0, 7, 3, 5, 6}
5
{0, 2, 3, 5, 6, 7}
6
{2, 0, 7, 3, 5, 6}
6
Average distance of nodes to root = 1.83333
7
Average distance of nodes to root = 1.83333
Average distance of nodes to root = 1.83333
Tree is not balanced.
{2, 0, 7, 3, 5, 6}
6
{0, 2, 3, 5, 6, 7}
Tree is not balanced.
{0, 2, 3, 5, 6, 7}
6
Tree is not balanced.
<5> is in tree.
<4> is not in tree.
5
{2, 0, 7, 3, 5, 6}
{2, 0, 7, 3, 5, 6}
{2, 0, 7, 3, 5, 6}
<7> is in tree.
Average distance of nodes to root = 1.83333
<0> is in tree.
Tree is not balanced.
Tree is not balanced.
{0, 2, 3, 5, 6, 7}
{2, 0, 7, 3, 5, 6}
{0, 2, 3, 5, 6, 7}
{0, 2, 3, 5, 6, 7}
{2, 0, 7, 3, 5, 6}
6
6
6
<5> is in tree.
Average distance of nodes to root = 1.83333
6
<5> is in tree.
{0, 2, 3, 5, 6, 7}
{2, 0, 7, 3, 5, 6}
Tree is not balanced.
Average distance of nodes to root = 1.83333
3
6
6
Tree is not balanced.
6
7
{0, 2, 3, 5, 6, 7}
6
{0, 2, 3, 5, 6, 7}
6
{2, 0, 7, 3, 5, 6}
6
Average distance of nodes to root = 1.83333
<5> is in tree.
Average distance of nodes to root = 1.83333
1
{0, 2, 3, 5, 6, 7}
Average distance of nodes to root = 1.83333
3
<1> is not in tree.
{2, 0, 7, 3, 5, 6}
<7> is in tree.
{2, 0, 7, 3, 5, 6}
Tree is not balanced.
Average distance of nodes to root = 1.83333
Average distance of nodes to root = 1.83333
Average distance of nodes to root = 1.83333
<4> is not in tree.
<1> is in tree.
Tree is not balanced.
<4> is not in tree.
3
Tree is not balanced.
<0> is in tree.
2
3
7
Tree is not balanced.
{0, 1, 2, 3, 5, 6, 7}
Average distance of nodes to root = 1.85714
{0, 1, 2, 3, 5, 6, 7}
7
<7> is in tree.
{0, 1, 2, 3, 5, 6, 7}
{2, 0, 7, 1, 3, 5, 6}
4
{0, 1, 2, 3, 5, 6, 7}
Average distance of nodes to root = 1.85714
{0, 1, 2, 3, 5, 6, 7}
Tree is not balanced.
Tree is not balanced.
<1> is in tree.
7
<0> is in tree.
<0> is in tree.
7
Average distance of nodes to root = 1.85714
Tree is not balanced.
Average distance of nodes to root = 1.85714
Tree is not balanced.
Tree is not balanced.
0
<3> is in tree.
Tree is not balanced.
Tree is not balanced.
<6> is in tree.
Tree is not balanced.
{0, 1, 2, 3, 5, 6, 7}
Tree is not balanced.
7
<2> is in tree.
1
8
<7> is in tree.
<1> is in tree.
{2, 0, 7, 1, 3, 5, 4, 6}
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
Tree is not balanced.
Tree is not balanced.
<3> is in tree.
8
5
Average distance of nodes to root = 2.125
8
8
4
Tree is not balanced.
5
8
<5> is in tree.
{2, 0, 7, 1, 3, 5, 4, 6}
Average distance of nodes to root = 2.125
Average distance of nodes to root = 2.125
8
Tree is not balanced.
Average distance of nodes to root = 2.125
<1> is in tree.
<5> is in tree.
{2, 0, 7, 1, 3, 5, 4, 6}
<4> is in tree.
7
<3> is in tree.
7
2
Tree is not balanced.
5
Tree is not balanced.
Tree is not balanced.
{2, 0, 7, 1, 3, 5, 4, 6}
6
<1> is in tree.
Average distance of nodes to root = 2.125
Tree is not balanced.
1
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.125
Average distance of nodes to root = 2.125
Tree is not balanced.
{2, 0, 7, 1, 3, 5, 4, 6}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
<6> is in tree.
8
2
{2, 0, 7, 1, 3, 5, 4, 6}
{0, 1, 2, 3, 4, 5, 6, 7}
4
<4> is in tree.
Tree is not balanced.
8
{2, 0, 7, 1, 3, 5, 4, 6}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Tree is not balanced.
8
1
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Average distance of nodes to root = 2.125
Average distance of nodes to root = 2.125
6
<1> is in tree.
<0> is in tree.
Average distance of nodes to root = 2.125
{2, 0, 7, 1, 3, 5, 4, 6}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
0
4
{2, 0, 7, 1, 3, 5, 4, 6}
Average distance of nodes to root = 2.125
{2, 0, 7, 1, 3, 5, 4, 6}
Average distance of nodes to root = 2.125
8
Average distance of nodes to root = 2.125
8
Tree is not balanced.
<7> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
<4> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.125
{2, 0, 7, 1, 3, 5, 4, 6}
8
8
Average distance of nodes to root = 2.125
{2, 0, 7, 1, 3, 5, 4, 6}
{2, 0, 7, 1, 3, 5, 4, 6}
{2, 0, 7, 1, 3, 5, 4, 6}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
{2, 0, 7, 1, 3, 5, 4, 6}
Tree is not balanced.
8
Average distance of nodes to root = 2.125
Tree is not balanced.
Average distance of nodes to root = 2.125
Tree is not balanced.
8
Average distance of nodes to root = 2.125
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Average distance of nodes to root = 2.125
Average distance of nodes to root = 2.125
<0> is in tree.
Average distance of nodes to root = 2.125
Average distance of nodes to root = 2.125
4
Tree is not balanced.
Average distance of nodes to root = 2.125
Average distance of nodes to root = 2.125
Average distance of nodes to root = 2.125
8
Average distance of nodes to root = 2.125
Average distance of nodes to root = 2.125
<6> is in tree.
<7> is in tree.
1
8
Average distance of nodes to root = 2.125
Average distance of nodes to root = 2.125
{0, 1, 2, 3, 4, 5, 6, 7}
8
3
6
<2> is in tree.
7
Tree is not balanced.
{2, 0, 7, 1, 3, 5, 4, 6}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
3
<7> is in tree.
<6> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
<5> is in tree.
Average distance of nodes to root = 2.125
8
Average distance of nodes to root = 2.125
{2, 0, 7, 1, 3, 5, 4, 6}
<6> is in tree.
{2, 0, 7, 1, 3, 5, 4, 6}
1
8
{2, 0, 7, 1, 3, 5, 4, 6}
1
4
8
3
Tree is not balanced.
Average distance of nodes to root = 2.125
6
{0, 1, 2, 3, 4, 5, 6, 7}
3
{0, 1, 2, 3, 4, 5, 6, 7}
<2> is in tree.
4
Average distance of nodes to root = 2.125
Average distance of nodes to root = 2.125
{2, 0, 7, 1, 3, 5, 4, 6}
{2, 0, 7, 1, 3, 5, 4, 6}
<5> is in tree.
8
{2, 0, 7, 1, 3, 5, 4, 6}
{2, 0, 7, 1, 3, 5, 4, 6}
Average distance of nodes to root = 2.125
5
3
{0, 1, 2, 3, 4, 5, 6, 7}
8
<2> is in tree.
{2, 0, 7, 1, 3, 5, 4, 6}
<1> is in tree.
2
Tree is not balanced.
Average distance of nodes to root = 2.125
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
<1> is in tree.
Average distance of nodes to root = 2.125
<6> is in tree.
<5> is in tree.
{2, 0, 7, 1, 3, 5, 4, 6}
4
<0> is in tree.
8
Tree is not balanced.
Tree is not balanced.
Average distance of nodes to root = 2.125
5
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
{2, 0, 7, 1, 3, 5, 4, 6}
8
{2, 0, 7, 1, 3, 5, 4, 6}
8
0
Average distance of nodes to root = 2.125
Average distance of nodes to root = 2.125
{0, 1, 2, 3, 4, 5, 6, 7}
<0> is in tree.
8
<3> is in tree.
1
5
8
Average distance of nodes to root = 2.125
Tree is not balanced.
6
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 0, 7, 1, 3, 5, 4, 6}
{2, 0, 7, 1, 3, 5, 4, 6}
8
{2, 0, 7, 1, 3, 5, 4, 6}
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 0, 7, 1, 3, 5, 4, 6}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.125
{2, 0, 7, 1, 3, 5, 4, 6}
Tree is not balanced.
{2, 0, 7, 1, 3, 5, 4, 6}
8
{2, 0, 7, 1, 3, 5, 4, 6}
Average distance of nodes to root = 2.125
2
<4> is in tree.
{2, 0, 7, 1, 3, 5, 4, 6}
Average distance of nodes to root = 2.125
{2, 0, 7, 1, 3, 5, 4, 6}
8
{2, 0, 7, 1, 3, 5, 4, 6}
8
3
4
8
Tree is not balanced.
<2> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
<0> is in tree.
Tree is not balanced.
8
{2, 0, 7, 1, 3, 5, 4, 6}
8
Tree is not balanced.
Average distance of nodes to root = 2.125
{2, 0, 7, 1, 3, 5, 4, 6}
Tree is not balanced.
{2, 0, 7, 1, 3, 5, 4, 6}
8
Tree is not balanced.
8
<2> is in tree.
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
8
Average distance of nodes to root = 2.125
1
Average distance of nodes to root = 2.125
{0, 1, 2, 3, 4, 5, 6, 7}
8
{2, 0, 7, 1, 3, 5, 4, 6}
{2, 0, 7, 1, 3, 5, 4, 6}
Average distance of nodes to root = 2.125
0
<3> is in tree.
<4> is in tree.
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
Average distance of nodes to root = 2.125
Average distance of nodes to root = 2.125
4
8
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
4
{2, 0, 7, 1, 3, 5, 4, 6}
<1> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
<0> is in tree.
{2, 0, 7, 1, 3, 5, 4, 6}
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 0, 7, 1, 3, 5, 4, 6}
<3> is in tree.
1
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
<7> is in tree.
{2, 0, 7, 1, 3, 5, 4, 6}
1
6
Average distance of nodes to root = 2.125
8
<3> is in tree.
8
8
Tree is not balanced.
Tree is not balanced.
{2, 0, 7, 1, 3, 5, 4, 6}
{2, 0, 7, 1, 3, 5, 4, 6}
Tree is not balanced.
<6> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.125
8
Tree is not balanced.
<4> is in tree.
<2> is in tree.
8
Tree is not balanced.
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
Tree is not balanced.
Tree is not balanced.
Average distance of nodes to root = 2.125
8
Tree is not balanced.
{2, 0, 7, 1, 3, 5, 4, 6}
<2> is in tree.
Average distance of nodes to root = 2.125
{2, 0, 7, 1, 3, 5, 4, 6}
8
4
<5> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.125
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 0, 7, 1, 3, 5, 4, 6}
4
Average distance of nodes to root = 2.125
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
4
5
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
{2, 0, 7, 1, 3, 5, 4, 6}
{2, 0, 7, 1, 3, 5, 4, 6}
8
Average distance of nodes to root = 2.125
{2, 0, 7, 1, 3, 5, 4, 6}
{2, 0, 7, 1, 3, 5, 4, 6}
7
<5> is in tree.
{2, 0, 7, 1, 3, 5, 4, 6}
5
{2, 0, 7, 1, 3, 5, 4, 6}
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 0, 7, 1, 3, 5, 4, 6}
{2, 0, 7, 1, 3, 5, 4, 6}
4
2
Average distance of nodes to root = 2.125
{2, 0, 7, 1, 3, 5, 4, 6}
Average distance of nodes to root = 2.125
0
{0, 1, 2, 3, 4, 5, 6, 7}
<5> is in tree.
<1> is in tree.
{2, 0, 7, 1, 3, 5, 4, 6}
Average distance of nodes to root = 2.125
<6> is in tree.
{2, 0, 7, 1, 3, 5, 4, 6}
{2, 0, 7, 1, 3, 5, 4, 6}
{2, 0, 7, 1, 3, 5, 4, 6}
Tree is not balanced.
Tree is not balanced.
<4> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 0, 7, 1, 3, 5, 4, 6}
Average distance of nodes to root = 2.125
Tree is not balanced.
8
8
8
Average distance of nodes to root = 2.125
<3> is in tree.
{2, 0, 7, 1, 3, 5, 4, 6}
0
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 0, 7, 1, 3, 5, 4, 6}
<5> is in tree.
Average distance of nodes to root = 2.125
4
{2, 0, 7, 1, 3, 5, 4, 6}
<3> is in tree.
<6> is in tree.
<5> is in tree.
{2, 0, 7, 1, 3, 5, 4, 6}
{0, 1, 2, 3, 4, 5, 6, 7}
<5> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 0, 7, 1, 3, 5, 4, 6}
<0> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
<3> is in tree.
8
Tree is not balanced.
<5> is in tree.
<2> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
<0> is in tree.
{2, 0, 7, 1, 3, 5, 4, 6}
Tree is not balanced.
8
<4> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
<7> is in tree.
8
Tree is not balanced.
Tree is not balanced.
8
{2, 0, 7, 1, 3, 5, 4, 6}
5
Average distance of nodes to root = 2.125
{2, 0, 7, 1, 3, 5, 4, 6}
{2, 0, 7, 1, 3, 5, 4, 6}
Average distance of nodes to root = 2.125
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 0, 7, 1, 3, 5, 4, 6}
<0> is in tree.
Tree is not balanced.
8
{2, 0, 7, 1, 3, 5, 4, 6}
Average distance of nodes to root = 2.125
8
8
Average distance of nodes to root = 2.125
2
Tree is not balanced.
<5> is in tree.
{2, 0, 7, 1, 3, 5, 4, 6}
Average distance of nodes to root = 2.125
{2, 0, 7, 1, 3, 5, 4, 6}
<3> is in tree.
<2> is in tree.
<6> is in tree.
{2, 0, 7, 1, 3, 5, 4, 6}
<6> is in tree.
Tree is not balanced.
Average distance of nodes to root = 2.125
<1> is in tree.
8
Average distance of nodes to root = 2.125
Average distance of nodes to root = 2.125
{2, 0, 7, 1, 3, 5, 4, 6}
Average distance of nodes to root = 2.125
Average distance of nodes to root = 2.125
Tree is not balanced.
{2, 0, 7, 1, 3, 5, 4, 6}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.125
Average distance of nodes to root = 2.125
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.125
4
{2, 0, 7, 1, 3, 5, 4, 6}
8
8
Average distance of nodes to root = 2.125
<3> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 0, 7, 1, 3, 5, 4, 6}
Average distance of nodes to root = 2.125
4
0
{0, 1, 2, 3, 4, 5, 6, 7}
5
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.125
8
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 0, 7, 1, 3, 5, 4, 6}
Tree is not balanced.
Tree is not balanced.
4
6
Average distance of nodes to root = 2.125
1
Average distance of nodes to root = 2.125
{2, 0, 7, 1, 3, 5, 4, 6}
<2> is in tree.
Average distance of nodes to root = 2.125
Tree is not balanced.
1
{2, 0, 7, 1, 3, 5, 4, 6}
{2, 0, 7, 1, 3, 5, 4, 6}
<3> is in tree.
<6> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.125
{0, 1, 2, 3, 4, 5, 6, 7}
7
Average distance of nodes to root = 2.125
6
8
{2, 0, 7, 1, 3, 5, 4, 6}
Tree is not balanced.
6
<5> is in tree.
5
{0, 1, 2, 3, 4, 5, 6, 7}
3
Average distance of nodes to root = 2.125
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 0, 7, 1, 3, 5, 4, 6}
Average distance of nodes to root = 2.125
{2, 0, 7, 1, 3, 5, 4, 6}
<0> is in tree.
Average distance of nodes to root = 2.125
{0, 1, 2, 3, 4, 5, 6, 7}
8
8
1
Average distance of nodes to root = 2.125
3
8
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
0
5
8
Average distance of nodes to root = 2.125
{2, 0, 7, 1, 3, 5, 4, 6}
3
{2, 0, 7, 1, 3, 5, 4, 6}
8
<3> is in tree.
8
Average distance of nodes to root = 2.125
{2, 0, 7, 1, 3, 5, 4, 6}
{0, 1, 2, 3, 4, 5, 6, 7}
8
<2> is in tree.
4
8
Average distance of nodes to root = 2.125
3
Tree is not balanced.
<0> is in tree.
4
Tree is not balanced.
Tree is not balanced.
Tree is not balanced.
{2, 0, 7, 1, 3, 5, 4, 6}
Tree is not balanced.
2
Average distance of nodes to root = 2.125
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.125
<4> is in tree.
Average distance of nodes to root = 2.125
<1> is in tree.
{2, 0, 7, 1, 3, 5, 4, 6}
Average distance of nodes to root = 2.125
Average distance of nodes to root = 2.125
Average distance of nodes to root = 2.125
Tree is not balanced.
<0> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Tree is not balanced.
{2, 0, 7, 1, 3, 5, 4, 6}
{2, 0, 7, 1, 3, 5, 4, 6}
<3> is in tree.
8
<1> is in tree.
8
Tree is not balanced.
Tree is not balanced.
Average distance of nodes to root = 2.125
{0, 1, 2, 3, 4, 5, 6, 7}
{2, 0, 7, 1, 3, 5, 4, 6}
8
1
8
Average distance of nodes to root = 2.125
8
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2.125
{2, 0, 7, 1, 3, 5, 4, 6}
0
4
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
<3> is in tree.
Average distance of nodes to root = 2.125
Tree is not balanced.
Tree is not balanced.
Average distance of nodes to root = 2.125
5
{0, 1, 2, 3, 4, 5, 6, 7}
